db/python/layers/analysis_runner.py

```diff
@@ -0,0 +1,40 @@
+from db.python.connect import Connection
+from db.python.layers.base import BaseLayer
+from db.python.tables.analysis_runner import AnalysisRunnerFilter, AnalysisRunnerTable
+from models.models.analysis_runner import AnalysisRunnerInternal
+
+
+class AnalysisRunnerLayer(BaseLayer):
+    """Layer for analysis logic"""
+
+    def __init__(self, connection: Connection):
+        super().__init__(connection)
+
+        self.at = AnalysisRunnerTable(connection)
+
+    # GETS
+
+    async def query(
+        self, filter_: AnalysisRunnerFilter, check_project_ids: bool = True
+    ) -> list[AnalysisRunnerInternal]:
+        """Get analysis runner logs"""
+        logs = await self.at.query(filter_)
+        if not logs:
+            return []
+
+        if check_project_ids:
+            project_ids = set(log.project for log in logs)
+
+            await self.ptable.check_access_to_project_ids(
```

Comment:
If I understand this correctly, if logs contains at least one project which user is not allowed to see, the whole query will throw exception, instead returning subset of logs user can see.
For instance if I look for runs in the last 24 hours, I might get exception instead of subset of runs for projects I am member of.

Reply:
Yes, but a caller can filter to the projects they have access to. So IMO sort of on them to filter for what they can see, rather than us applying a hidden filtering - FWIW we don't do this anywhere except the "my projects" API call.